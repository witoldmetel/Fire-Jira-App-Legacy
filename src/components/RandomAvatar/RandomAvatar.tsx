import React from 'react';
import classnames from 'classnames';

import { getAvatarImage } from '../../utils/helpers';

export interface Props {
  randomFace: string;

  className?: string;

  onMouseEnter?: () => void;
  onMouseLeave?: () => void;
}

export class RandomAvatar extends React.Component<Props> {
  private get className() {
    return classnames('ui mini circular image', this.props.className);
  }

  private get imgUrl() {
    return getAvatarImage(this.props.randomFace);
  }

  public render() {
    return (
      <img
        src={this.imgUrl}
        className={this.className}
        onMouseEnter={this.props.onMouseEnter}
        onMouseLeave={this.props.onMouseLeave}
      />
    );
  }
}
