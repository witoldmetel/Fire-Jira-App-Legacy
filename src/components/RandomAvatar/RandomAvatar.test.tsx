import * as React from 'react';
import { shallow } from 'enzyme';

import { RandomAvatar } from './RandomAvatar';

describe('<RandomAvatar>', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<RandomAvatar randomFace={'randomFace'} />);

    expect(wrapper).toMatchSnapshot();
  });
});
