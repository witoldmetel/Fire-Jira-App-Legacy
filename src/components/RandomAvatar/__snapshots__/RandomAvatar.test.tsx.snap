// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RandomAvatar> renders correctly 1`] = `
<img
  className="ui mini circular image"
  src="https://api.hello-avatar.com/adorables/avatars/randomFace.png"
/>
`;
